spec.bs

```diff
@@ -1226,6 +1254,280 @@ On the other hand, methods for getting data from the [=shared storage database=]
+                     1. Continue.
+              <dt> If |operationString| is "`delete`":
+                <dd> Perform the following steps:
+                     1. Let |key| be the result of running [=obtain a string-like parameter value=] with |parameters| and "`key`".
+                     1. If |key| is not null, run {{WindowSharedStorage/delete()}} on |sharedStorage| with |key|.
+                     1. Continue.
+              <dt> If |operationString| is "`append`":
+                <dd> Perform the following steps:
+                     1. Let |key| be the result of running [=obtain a string-like parameter value=] with |parameters| and "`key`".
+                     1. If |key| is null, continue.
+                     1. Let |value| be the result of running [=obtain a string-like parameter value=] with |parameters| and "`value`".
+                     1. If |value| is not null, run {{WindowSharedStorage/append()}} on |sharedStorage| with |key| and |value|.
+                     1. Continue.
+              <dt> If |operationString| is "`set`":
+                <dd> Perform the following steps:
+                     1. Let |key| be the result of running [=obtain a string-like parameter value=] with |parameters| and "`key`".
+                     1. If |key| is null, continue.
+                     1. Let |value| be the result of running [=obtain a string-like parameter value=] with |parameters| and "`value`".
+                     1. If |value| is null, continue.
+                     1. Let |options| be a new {{SharedStorageSetMethodOptions}}.
+                     1. If the result of running [=obtain a boolean parameter value=] with |parameters| and "`ignore_if_present`" is true, [=map/set=] |options|["`ignoreIfPresent`"] to true..
+                     1. Run {{WindowSharedStorage/set()}} on |sharedStorage| with |key|, |value|, and |options|.
+                     1. Continue.
+              <dt> If |operationString| is anything else:
+                <dd> Continue.
+            </dl>
+  </div>
+
+  <div algorithm>
+    To <dfn>check if string-like</dfn> for a [=Bare Item=] |item|, do the following:
+    1. If the |item| is a [=item/String=], [=item/Token=], or [=item/Byte Sequence=], return true.
+    1. Otherwise, return false.
+  </div>
+
+  <div algorithm>
+    To <dfn>get the string value</dfn> for a [=Bare Item=] |item|, do the following:
+    1. If the result of running [=check if string-like=] on |item| is false, return failure.
+    1. Let |itemAsString| be the |item|'s underlying [=infra/string=].
+    1. [=Assert=] that |itemAsString| is an [=ASCII string=].
+    1. Switch on the type of |item|:
+        <dl class=switch>
+          <dt> If |item| is a [=item/Token=]:
+            <dd> Return |itemAsString|.
+          <dt> If |item| is a [=item/String=]:
+            <dd> Return the result of [=unwrap a delimited ASCII string|unwrapping a delimited ASCII string=] for |itemAsString| and the double-quote character.
```

Comment:
This would be incorrect because of escaping, but fortunately https://fetch.spec.whatwg.org/#concept-header-list-get-structured-header already converted these to strings.

Reply:
Done